pkg/api/controllers/volume.go

```diff
@@ -66,21 +66,45 @@ func (v *VolumePortal) CreateVolume() {
 	// get profile
 	var prf *model.ProfileSpec
 	var err error
-	if volume.ProfileId == "" {
-		log.Warning("Use default profile when user doesn't specify profile.")
-		prf, err = db.C.GetDefaultProfile(ctx)
-		// Assign the default profile id to volume so that users can know which
-		// profile is used for creating a volume.
-		volume.ProfileId = prf.Id
+	// If pool id is not specified, get a profile or take default pool
+	if volume.PoolId == "" {
+		if volume.ProfileId == "" {
+			log.Warning("use default profile when user doesn't specify profile.")
+			prf, err = db.C.GetDefaultProfile(ctx)
+			// Assign the default profile id to volume so that users can know which
+			// profile is used for creating a volume.
+			if err == nil {
+				volume.ProfileId = prf.Id
+
+			} else {
+				//return Error: user should either have Profile or PoolID
+				errMsg := fmt.Sprintf("no pool id in the request or no default profile available. %s", err.Error())
+				v.ErrorHandle(model.ErrorBadRequest, errMsg)
+				return
+
+			}
+		} else {
+			prf, err = db.C.GetProfile(ctx, volume.ProfileId)
+		}
+		if err != nil {
+			errMsg := fmt.Sprintf("get profile failed: %s", err.Error())
+			v.ErrorHandle(model.ErrorBadRequest, errMsg)
+			return
+		}
+	} else if volume.ProfileId != "" {
+		// if both PoolID and ProfileID is given then use only PoolID
+		volume.ProfileId = ""
+		log.Warning("using only poolid when user provided both pool and profile .")
 	} else {
-		prf, err = db.C.GetProfile(ctx, volume.ProfileId)
-	}
-	if err != nil {
-		errMsg := fmt.Sprintf("get profile failed: %s", err.Error())
-		v.ErrorHandle(model.ErrorBadRequest, errMsg)
-		return
+		pool, err := db.C.GetPool(ctx, volume.PoolId)
```

Comment:
One doubt, If user doesn't add the pool details in DB, then GetPool will fail. Now next question is, how user will add the pool details into db?

Reply:
As discussed, As of now pool is a prerequiste for any operation in hotpot. 
pool creation is in the backend and hotpot only holds metadata of the pool. 
we can think of having create(POST ) /pool operation in hotpot.